packages/lesmis-server/src/routes/ReportRoute.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ *
+ */
+
+import { Request, Response, NextFunction } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+import { ReportConnection, WebConfigConnection } from '../connections';
+
+export class ReportRoute extends Route {
+  private readonly reportConnection: ReportConnection;
+  private readonly webConfigConnection: WebConfigConnection;
+
+  constructor(req: Request, res: Response, next: NextFunction) {
+    super(req, res, next);
+
+    this.reportConnection = new ReportConnection(req.session);
+    this.webConfigConnection = new WebConfigConnection(req.session);
+  }
+
+  async buildResponse() {
+    const { reportCode } = this.req.params;
+    const { type } = this.req.query;
+    switch(type) {
```

Comment:
Are there differences in the parameters required by the three? Ideally that's the domain of the orchestration server, so the lesmis front end doesn't have to deal with constructing different types of url. However also happy to keep it as is if @tcaiger has already made lesmis front end do exactly that!

Reply:
There are different parameters required, I wrote it dumb because I thought we already had those urls generated on the front end. Will get Tom to confirm, and will add smarts if needed.

Reply:
After a chat with Tom I've added some extra handling to account for some of the shared parameters between these endpoints, the front end may still pass through extra query parameters in cases where they're needed